### DISCUSSION TITLE
Using `math.random()` in css module leads to a wrongly generated class names

### DISCUSSION THREAD

**Discussion by orangecoloured** [preactjs/preact#4238] [Q&A]
I used `create-preact` to initialise my app.
It uses `vite v4.3.2` and `sass v1.69.5`.

I have a scss module with a code that contains `math.random()`.

```
.someClass {
    some-value: 100px + math.random(1000);
}
```

And because of the `math.random()` as a result the names that are assigned to the html elements are not that are being generated and loaded in the stylesheet.

With the `math.random()`
<img width="401" alt="Screenshot 2023-12-19 at 11 18 02 PM" src="https://github.com/vitejs/vite/assets/3314891/8e8afed0-f156-42de-9f6e-f39248e6e9bf">

Without the `math.random()`
<img width="401" alt="Screenshot 2023-12-19 at 11 17 39 PM" src="https://github.com/vitejs/vite/assets/3314891/cc3c500c-b564-438e-bec0-ab50e3c50683">

For clarity. When the `math.random()` is used the generated class name that is used for `class=""` has `1lzbo` part in the class name, but in the injected <style> tag it's `yl1g2`. When the `math.random()` is not used, the class names are correct. Both have `1u0g5` in their name.

The css property name that is using `math.random()` does not matter. The behaviour is consistent per file which contains `math.random()` while `npm run dev`.

I tried reproducing it, but no success. The minimal version works correctly.
https://stackblitz.com/edit/vitejs-vite-dtieff?file=style.module.scss

Haven't yet tried upgrading `vite` to the latest version.

What could be wrong? Which part could be failing here? And how can I try to fix this? Apart from not using `math.random()`.

**Comment by marvinhagemeister** [comment]
This sounds like a problem with `sass` or `vite`, not with Preact itself. Please file an issue to the appropriate issue tracker.
